### ISSUE TITLE
The mv command performs a copy even if the source and destination are on the same device.

### ISSUE THREAD

**Issue by hamflx** [nushell/nushell#12015]
### Describe the bug

```nushell
C:\repos\my-project> mv node_modules\ \.nd\a13
Error:   × Could not move "C:\\repos\\my-project\\node_modules\\" to "C:\\.nd\\a13". Error Kind: The system cannot find the path specified. (os error 3)
   ╭─[entry #2:1:1]
 1 │ mv node_modules\ \.nd\a13
   ·    ──────┬──────
   ·          ╰── could not move
   ╰────

C:\repos\my-project> timeit {mv node_modules\ \.nd\a13}
Error:   × Could not move "C:\\repos\\my-project\\node_modules\\" to "C:\\.nd\\a13". Error Kind: The system cannot find the path specified. (os error 3)
   ╭─[entry #7:1:1]
 1 │ timeit {mv node_modules\ \.nd\a13}
   ·            ──────┬──────
   ·                  ╰── could not move
   ╰────


1min 58sec 687ms 682µs 600ns
```

### How to reproduce

1. Clone a large front-end project.
2. Run `pnpm install`.
3. `cd node_modules` to simulate a file being locked by a process.
4. Open a new shell.
5. Run `mv node_modules \.nd`. This command will take a long time and then fail.

### Expected behavior

I expect the `mv` command failed immediately instead of after copy operation.

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.90.2                                   |
| branch             | main                                     |
| commit_hash        | f24877ba081f754251188f902f1bed18aa270bcf |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.74.1 (a28077b28 2023-12-04)      |
| rust_channel       | 1.74.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.74.1 (ecb9851af 2023-10-18)      |
| build_time         | 2024-02-29 11:23:06 +08:00               |
| build_rust_channel | debug                                    |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

and

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.89.0                                   |
| branch             |                                          |
| commit_hash        | 2c1560e281d65a50d90b30c42006755b364b4928 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.73.0 (cc66ad468 2023-10-03)      |
| rust_channel       | 1.73.0-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.73.0 (9c4383fb5 2023-08-26)      |
| build_time         | 2024-01-09 20:28:44 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
If there are sharing violations in Windows, you won't be able to move files. This is a Windows thing, not a nushell thing. Also, we're about to stop using `mv` altogether and replace it with the coreutils `umv` command (and we'll rename it to mv).

**Comment by hamflx** [comment]
@fdncred 

> If there are sharing violations in Windows, you won't be able to move files.

You are right. but in this scenario, `mv`/`umv` still try to copy files, and then failed, this takes a long time, and can not be stopped by ctrl-c.

**Comment by hamflx** [comment]
on Windows, deleting plenty of small files is too slow, so I have using `mv` instead of `rm` to delete `node_modules`, but it was takes a long time (several minutes, cannot be stopped by ctrl-c https://github.com/nushell/nushell/issues/10144#issuecomment-1970713067), and then failed, this makes me annoying. so i think this can be optimized.

**Comment by fdncred** [comment]
I commented on the best strategy right now for mv in your PR.
